docs/resources/azurerm_subnet.md.erb

```diff
@@ -0,0 +1,153 @@
+platform: azure
+---
+
+# azurerm\_subnet
+
+Use the `azurerm_subnet` InSpec audit resource to test properties related to a
+subnet for a given virtual network.
+
+<br />
+
+## Availability
+
+### Installation
+
+This resource is available in the `inspec-azure` [resource
+pack](https://www.inspec.io/docs/reference/glossary/#resource-pack). To use it, add the
+following to your `inspec.yml` in your top-level profile:
+
+    depends:
+      inspec-azure:
+        git: https://github.com/inspec/inspec-azure.git
+
+You'll also need to setup your Azure credentials; see the resource pack
+[README](https://github.com/inspec/inspec-azure#inspec-for-azure).
+
+### Version
+
+This resource first became available in 1.2.0 of the inspec-azure resource pack.
+
+## Syntax
+
+The `resource_group`, virtual network `vnet`, and subnet 'name' must be given as parameters.
+Since subnet must be in the context of a virtual network, this parameter had to be included.
+
+    describe azurerm_subnet(resource_group: 'MyResourceGroup', vnet: 'MyVnetName', name: 'MySubnetName') do
+      ...
+    end
+
+<br />
+
+## Examples
+### Ensure that the subnet exists for a given virtual network.
+    describe azurerm_subnet(resource_group: 'MyResourceGroup', vnet: 'MyVnetName', name: 'MySubnetName') do
+      it { should exist }
+    end
+
+### Ensure that the subnets address prefix is configured as expected.
+    describe azurerm_subnet(resource_group: 'MyResourceGroup', vnet: 'MyVnetName', name: 'MySubnetName') do
+        its('address_prefix') { should eq [192.168.0.0/24] }
+    end
+
+### Ensure that the subnet is attached to the right network security group.
+    describe azurerm_subnet(resource_group: 'MyResourceGroup', vnet: 'MyVnetName', name: 'MySubnetName') do
+        its('nsg') { should eq 'NetworkSecurityGroupName'}
+    end
+
+<br />
+
+## Parameters
+
+  - `resource_group`, `vnet`, `name`
+
+## Parameter Examples
+
+### resource\_group (required)
+
+Defines the resource group of the subnet that you wish to test resides in.
+
+    describe azurerm_subnet(resource_group: 'MyResourceGroup', vnet: 'MyVnetName', name: 'MySubnetName') do
+      ...
+    end
+
+### vnet (required)
+
+Defines the virtual network that the subnet that you wish to test is a part of.
+
+    describe azurerm_subnet(resource_group: 'MyResourceGroup', vnet: 'MyVnetName', name: 'MySubnetName') do
+      ...
+    end
+
+
+### name (required)
+
+Defines the name of the subnet that you wish to test.
+
+    describe azurerm_subnet(resource_group: 'MyResourceGroup', vnet: 'MyVnetName', name: 'MySubnetName') do
+      ...
+    end
+
+## Properties
+
+  - `id`, `name`, `type`, `address_prefix`, `nsg`
+
+### id
+
+The subnet's id.
+
+    its('id') { should eq(id) }
+
+Id will be in
+    format:
+
+    '/subscriptions/5dbd1a2e-7fd6-4891-9e3b-74f665be76d5/resourceGroups/Inspec-Azure-mmclane/providers/Microsoft.Network/virtualNetworks/Inspec-VNet/subnets/Inspec-Subnet'
```

Comment:
You should obfuscate your subscription id. I'd recommend being paranoid even though I don't think the subscription id is necessarily sensitive.